### DISCUSSION TITLE
mg_http_get_header_var() question

### DISCUSSION THREAD

**Discussion by pmeskoni** [cesanta/mongoose#1782] [Ask questions, get answers] [answered]
The http data i'm receiving in my application contains a header var in a file upload that looks like this:
<stuff>
headervar1=value1;headervar2=value2

<file content>
<stuff>

My question is - Is my http non-compliant, or should mg_http_get_header_var() also be looking for '\r', maybe '\n' when looking for the end of the value:
      while (p < x &&
             (q ? p == b || *p != '"' : *p != ';' && *p != ' ' && *p != ','))
becomes
      while (p < x &&
             (q ? p == b || *p != '"' : *p != ';' && *p != ' ' && *p != ',' && *p != '\r'))
??

**Comment by scaprile** [comment, marked as answer]
Do you mean HTTP headers ? No, you are not sending standard headers https://www.rfc-editor.org/rfc/rfc2616#page-31
Check the docs, tutorials, and examples; the [file upload tutorial](https://mongoose.ws/tutorials/file-uploads/)/examples show how to do something like what I think you want to do
